### PR TITLE
Update OS names to be more generic in kernel versions 10.0.x

### PR DESCRIPTION
**Pull request by bwatters-r7**
Previously, when Microsoft released a new "named" operating system, it bumped the major/minor kernel version.  Now, it is changing OS names based on the build version (or nothing).  That means we fingerprint Windows 2019, Server 1809, Server 1803, and Server 1709 as Windows 2016, as all are running some flavor of kernel 10.0.x.  Rather than trying to track build numbers (which would not always help, as Windows Server 2019 shares a build number with Windows Server Release 1809) this just makes the OS name for any Kernel version 10.0.x as "Windows Server 2016+" and gives the version and build number.  As Microsoft continues to shrink the update and delivery cycle, the build number will be more important than the OS name.

I tried to play as little as possible in the output, in case some regex out there depended on it.  On a second though, it might be beneficial to preface the version with a 'version' tag, but I don't know how anyone else feels.

Fixes https://github.com/rapid7/metasploit-framework/issues/12112

Windows 7x64 SP1:
```
meterpreter > sysinfo
Computer        : WIN7X64SP1
OS              : Windows 7 (6.1 Build 7601, Service Pack 1).
Architecture    : x64
System Language : en_US
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x64/windows
```

Windows 2016:
```
meterpreter > sysinfo
Computer        : WIN2016X64
OS              : Windows 2016+ (10.0 Build 14393).
Architecture    : x64
System Language : en_US
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x64/windows
```

Windows 2019:
```
meterpreter > sysinfo
Computer        : WIN2019X64
OS              : Windows 2016+ (10.0 Build 17763).
Architecture    : x64
System Language : en_US
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x64/windows

```

Windows 2012:
```
meterpreter > sysinfo
Computer        : WIN2012X64
OS              : Windows 2012 (6.2 Build 9200).
Architecture    : x64
System Language : en_US
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x64/windows
```

Windows Server 1709:
```
meterpreter > sysinfo
Computer        : WIN1709X64
OS              : Windows 2016+ (10.0 Build 16299).
Architecture    : x64
System Language : en_US
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x64/windows
```

Windows Server 1803:
```
meterpreter > sysinfo
Computer        : WIN1709X64
OS              : Windows 2016+ (10.0 Build 16299).
Architecture    : x64
System Language : en_US
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x64/windows
```

Windows Server 1809:
```
meterpreter > sysinfo
Computer        : WIN1809X64
OS              : Windows 2016+ (10.0 Build 17763).
Architecture    : x64
System Language : en_US
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x64/windows

Windows 10x64 1809:
```
meterpreter > sysinfo
Computer        : WIN10X64_1809
OS              : Windows 10 (10.0 Build 17763).
Architecture    : x64
System Language : en_US
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x64/windows
```

### CODE REVIEW

**Comment by mkienow-r7** [comment]
Minor: near the top of the `add_windows_os_version` method the field you format using `%lu` are formatted using `%u`. Should those also be updated to use `%lu`?
```
dprintf("[VERSION] Major   : %u", v.dwMajorVersion);
dprintf("[VERSION] Minor   : %u", v.dwMinorVersion);
dprintf("[VERSION] Build   : %u", v.dwBuildNumber);
```

**Comment by mkienow-r7** [comment]
> On a second though, it might be beneficial to preface the version with a 'version' tag, but I don't know how anyone else feels.
Would you provide an example?

**Comment by bwatters-r7** [comment]
@mkienow-r7 that's a great catch about the long v. unsigned long.  Unfortunately, it is a bit of an ugly and long story:
I wrote the code based off existing code, which used unsigned longs for format.  That made sense because when I checked the function that populated it, `RtlGetVersion`,  MSDN (https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/content/wdm/nf-wdm-rtlgetversion) says it takes a pointer to a `OSVERSIONINFOEXW` struct (https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/content/wdm/ns-wdm-_osversioninfoexw) that is defined in `wdm.h` and populated with unsigned longs.  
When I checked the imports in meterpreter, it turns out were using the non-redirected `RtlGetVersion` redirection, so we're calling `RtlGetVersionA` which uses `OSVERSIONINFOEXW` (https://docs.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-osversioninfoexw) from WinNT.h that uses `DWORD`S, not `ulongs`.

TL;DR: We need to change the prints in the `_snprintf` to `%u` because we're using the older `RtlGetVersion`

**Comment by bwatters-r7** [comment]
> Would you provide an example?

"Tag" may be a little extreme, rather than :
`OS              : Windows 2016+ (10.0 Build 14393).`
Maybe something like this that would parse easier:
`OS              : Windows 2016+ (Kernel 10.0 Build 14393).`

**Comment by mkienow-r7** [comment]
I could see the labeling making it easier for regex processing. I'm not sure where the regex that depend on it are located, but it might be nice the code that was doing that was able to use a data object that represents all of the version elements.
